lib/lf/quad/make_quad_rule.h

```diff
@@ -26,4 +26,25 @@ namespace lf::quad {
  * - For Quadrilaterals it uses tensor products of Gauss-Legendre rules
  */
 QuadRule make_QuadRule(base::RefEl ref_el, unsigned char order);
+
+/** @brief edge midpoint quadrature rule for reference triangles
+ *
+ * This quadrature rule relies on point evaluations at the midpoints of
+ * all edges with equal weights = 1/6 for the reference triangle.
+ *
+ * The rule is exact for quadratic bi-variate polynomials
+ */
+QuadRule make_TriaQR_EdgeMidpointRule();
+
+/** @brief edge midpoint quadrature rule for unit square (= reference quad)
+ *
+ * This quadrature rule relies on point evaluations at the midpoints of
+ * all edges with equal weights = 1/4 for the unit square.
+ *
+ * The rule is exact for bilinear polynomials.
+ */
+QuadRule make_QuadQR_EdgeMidpointRule();
```

Comment:
I think it's often the case that we want to call `make_X_EdgeMidpointRule` but we don't know whether `X` is a triangle or quadrilateral. Of course the surrounding code could always make an if statement, but we also don't loose anything by merging `make_QuadQR_EdgeMidpointRule` and `make_TriaQR_EdgeMidpointRule` into `make_EdgeMidpointRule(RefEl ref_el)`? For `ref_el=Point` we could of course throw an error...